### ISSUE TITLE
W40 ("Security Groups egress with an IpProtocol of -1 found") should not trigger when CidrIp/CidrIpv6 is localhost

### ISSUE THREAD

**Issue by candrews** [stelligent/cfn_nag#289]
Using `IpProtocol` `-1` and `CidrIp` `127.0.0.1/32` or `CidrIpv6` `::1/128` is the recommended way  of removing the default egress rule (see "Remove Default Rule" https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group.html#aws-properties-ec2-security-group--seealso) so at all egress can be denied, for example:
```
"DBSecurityGroup": {
	"Type": "AWS::EC2::SecurityGroup",
	"Properties": {
		"GroupDescription": "Open database for access",
		"VpcId": {
			"Ref": "VPC"
		},
		"SecurityGroupEgress": [
			{
				"IpProtocol": "-1",
				"CidrIpv6": "::1/128"
			},
			{
				"IpProtocol": "-1",
				"CidrIp": "127.0.0.1/32"
			}
		]
	}
},
```

Currently, `cfn_nag` reports a `W40` warning on this fragment and it should not do so.

**Comment by Mr-Lizard** [comment]
I'm currently working on this...just having a spat with Rubocop, will submit once I am victorious :)

**Comment by erickascic** [comment]
hey @Mr-Lizard are you still working on this?  if you had work in this direction but didn't finish - would still love for you to submit it.  may have a resource to take on this issue.

**Comment by erickascic** [comment]
@candrews fixed in 0.5.21
